### PR TITLE
gh-92128: Add `__class_getitem__` to `logging.LoggerAdapter` and `logging.StreamHandler`

### LINKED ISSUE #92128: Add `__class_getitem__` to `logging.LoggerAdapter` and `logging.StreamHandler`
**Feature or enhancement**

`logging.LoggerAdapter` and `logging.StreamHandler` are generic in typeshed but cannot currently be parameterised at runtime. This has been reported several times to typeshed as being quite painful for end users, since these are both classes that users like to subclass:

- https://github.com/python/typeshed/issues/5680#issuecomment-1004762713
- https://github.com/python/typeshed/pull/5954#issuecomment-1114253234

If you're a user of type hints, subclassing a class that's generic in the stub but not at runtime usually means that you have to do something horrible with `if TYPE_CHECKING` (e.g. see https://github.com/python/typeshed/pull/5954#issuecomment-1114270968).

The stubs for these classes have been fairly stable in typeshed for a while; I think it's unlikely that we'll ever want to make them non-generic again over at typeshed. I therefore propose that `__class_getitem__` be added to these two classes in 3.11, so that they can be parameterised at runtime.

Cc. @vsajip for the `logging` module.

### PR DESCRIPTION
**Pull request by AlexWaygood**
Closes #92128

### CODE REVIEW

**Comment by AlexWaygood** [comment]
Previous changes like this have not made changes to the docs, but I'm happy to do so here if it's desirable.

**Review by JelleZijlstra — APPROVED** [review]
Looks good, I checked the stubs and it makes sense that these classes are generic.

I do think we should document that these classes are generic and say how many type arguments to expect and what they mean. We should do that more broadly for generic classes, though, and I'd rather get the runtime change in before the feature freeze.

**Review by vsajip — APPROVED** [review]

File: Misc/NEWS.d/next/Library/2022-05-01-21-45-41.gh-issue-92128.Di7VbE.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Add :meth:`~object.__class_getitem__` to :class:`logging.LoggerAdapter` and
```
Comment (kumaraditya303) [inline Misc/NEWS.d/next/Library/2022-05-01-21-45-41.gh-issue-92128.Di7VbE.rst:1]: Maybe add that it makes it generic.

**Review by kumaraditya303 — APPROVED** [review]

File: Misc/NEWS.d/next/Library/2022-05-01-21-45-41.gh-issue-92128.Di7VbE.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Add :meth:`~object.__class_getitem__` to :class:`logging.LoggerAdapter` and
+:class:`logging.StreamHandler`. Patch by Alex Waygood.
```
Comment (AlexWaygood) [inline Misc/NEWS.d/next/Library/2022-05-01-21-45-41.gh-issue-92128.Di7VbE.rst]:
```suggestion
:class:`logging.StreamHandler`, allowing them to be parameterized at runtime.
Patch by Alex Waygood.
```

**Review by kumaraditya303 — APPROVED** [review]

**Pull request merged**